validphys2/src/validphys/overfit_metric.py

```diff
@@ -0,0 +1,217 @@
+"""
+overfit_metric.py
+
+This module contains the functions used to calculate the overfit metric and 
+produce the corresponding tables and figures.
+"""
+
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+
+from reportengine import collect
+from reportengine.figure import figure
+from reportengine.table import table
+
+from validphys.checks import check_at_least_two_pdfs
+
+log = logging.getLogger(__name__)
+
+preds = collect(
+    "predictions",
+    (
+        "pdfs",
+        "dataset_inputs",
+    ),
+)
+
+
+def _create_new_val_pseudodata(pdf_data_index, fit_data_indices_list):
+    """Loads all validation psuedodata replicas used during the fiting of the
+    pdf replicas
+
+    Returns
+    -------
+    np.ndarray
+        (nrep,ndata) sized numpy array containingt the validation data used to
+        fit the pdfs.
+    """
+    vl_data_fitrep = []
+    for fitreplica_info in fit_data_indices_list:
+        vl_data_fitrep.append(
+            fitreplica_info.pseudodata.loc[pdf_data_index.val_idx]
+        )
+    return np.array(vl_data_fitrep)[:, :, 0]
+
+
+@check_at_least_two_pdfs
+def calculate_chi2s_per_replica(
+    recreate_pdf_pseudodata_no_table,
+    preds,
+    dataset_inputs,
+    groups_covmat_no_table,
+):
+    """Calculates the chi2 for each PDF relica to many pseudodata replicas.
```

Comment:
```suggestion
    """Calculates, for each PDF replica, the chi2 of the validation with the pseudodata generated
    for all other replicas in the fit
```

I've changed this a bit but please add more stuff if you wish. I think this is the "meat" of the method so I think it should be very precisely described.